boot/seal.go

```diff
@@ -506,12 +501,21 @@ func recoveryBootChainsForSystems(systems []string, trbl bootloader.TrustedAsset
 	return chains, nil
 }
 
-func runModeBootChains(rbl, bl bootloader.Bootloader, model *asserts.Model, modeenv *Modeenv, cmdlines []string) ([]bootChain, error) {
+func runModeBootChains(rbl, bl bootloader.Bootloader, model *asserts.Model, modeenv *Modeenv) ([]bootChain, error) {
 	tbl, ok := rbl.(bootloader.TrustedAssetsBootloader)
 	if !ok {
 		return nil, fmt.Errorf("recovery bootloader doesn't support trusted assets")
 	}
-
+	cmdlines := []string(modeenv.CurrentKernelCommandLines)
+	if len(cmdlines) == 0 {
+		// XXX: do we need this fallback for when the boot command lines
+		// in modeenv have not been set yet?
```

Comment:
in theory we should hit this only if we do a reseal before going through the mark succesful code from the last PR? shouldn't happen but maybe better to be safe anyway?  otoh this code should be somehow in resealKeyToModeenv instead

Reply:
Moved the code around a bit.